sdk/keyvault/azure-security-keyvault-secrets/tspconfig.yaml

```
@@ -18,4 +18,5 @@ options:
   "@azure-tools/typespec-cpp":
     flavor: azure
     package-dir: "azure-security-keyvault-secrets"
-    namespace: "Azure::Security::KeyVault::Secrets"
+    namespace: "Azure::Security::KeyVault::Secrets::_detail"
+    headersOutputPath: "azure/keyvault/secrets/generated"
```

So far we've always put both `_internal` and `_detail` namespace headers under `/internal`. Plus, if you use that directory, codegen has special logic to not produce an all-inclusive header, and will use the correct dll_import_export - not the one that it creates as a duplicate. 
```suggestion
    headersOutputPath: "azure/keyvault/secrets/internal"
```

---

per chat